### ISSUE TITLE
Use In-cluster Devfile registries for integration tests

### ISSUE THREAD

**Issue by valaparthvi** [redhat-developer/odo#6702]
/kind user-story

## User Story
As an odo developer, I want to use stable Devfile registries (but not the default Production one) for the integration tests So that any changes to Staging Registry (what we're currently using) do not break or block us from continuing odo development.

## Acceptance Criteria 
- [ ] It should setup an in-cluster Devfile registry. See https://odo.dev/blog/deploying-and-using-in-cluster-devfile-registry#deploying-a-devfile-registry-in-the-cluster
- [ ] Tests should be configured such that they use the staging registry by default, because it is still great that `odo` can detect and report such issues to the Devfile Services team.
- [ ] But if the tests fail due to the Staging Registry, it should fallback to using in-cluster registry. This can be done quite easily now that `odo` can automatically detect and use registries declared in the cluster. See https://odo.dev/blog/deploying-and-using-in-cluster-devfile-registry#declaring-the-devfile-registry

## Links 
- Related Epic (mandatory): 

/kind user-story

**Comment by feloy** [comment]
We have succesfully installed a Devfile Registry on the Kubernetes cluster on IBM Cloud.

The only tag available for images is `next`, which is the one where the problem happens.

We have been able to use a previous image saved by Parthvi but which is customized for specific tests, we will need to build an image from original sources.

The PR #6705 makes the tests accept a DEVFILE_REGISTRY env variable to point to this specific devfile registry; this env variable has been added to all CI jobs.
